lib/src/collection/bimap.dart

```diff
@@ -87,29 +87,11 @@ class HashBiMap<K, V> implements BiMap<K, V> {
   }
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_getter
-  Iterable<Null> get entries {
-    // Change Iterable<Null> to Iterable<MapEntry<K, V>> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("entries");
-  }
-
-  BiMap<V, K> get inverse {
-    if (_cached == null) {
-      _cached = new HashBiMap._from(_inverse, _map);
-    }
-    return _cached;
-  }
+  Iterable<MapEntry<K, V>> get entries => _map.entries;
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
-  Map<K2, V2> map<K2, V2>(Object transform(K key, V value)) {
-    // Change Object to MapEntry<K2, V2> when
-    // the MapEntry class has been added.
-    throw new UnimplementedError("map");
-  }
+  Map<K2, V2> map<K2, V2>(MapEntry<K2, V2> transform(K key, V value)) =>
+      _map.map(transform);
```

Comment:
Does `Map.map` document the expected return type? If it's expected to be a `BiMap`, we need to apply the invariants from `_add` (i.e. that not only must the keys be unique, but so must the values, since they're the keys of the inverse).

If it's not, then I think this is fine as-is.

Reply:
[It doesn't document really.](https://api.dartlang.org/dev/2.0.0-dev.37.0/dart-core/Map/map.html)

My thought is it looks just like `List.map`, Set.map`, ...: you don't expect a List or a Set out the other side; you just expect an Iterable.

Reply:
Cool. That sounds perfectly reasonable to me.